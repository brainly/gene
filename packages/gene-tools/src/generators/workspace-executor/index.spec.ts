import { Tree } from '@nrwl/devkit';
import * as devkit from '@nrwl/devkit';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
import workspaceExecutorGenerator from './index';

describe('workspaceExecutor generator', () => {
  let tree: Tree;
  const executorName = 'fooBar';
  const expectedExecutorName = 'foo-bar';

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace({layout: 'apps-libs'});

    tree.write(
      'tools/executors/executors.json',
      JSON.stringify({ executors: {} })
    );
    tree.write(
      'tools/executors/README.md',
      `# Executors

Some description

## Available workspace executors

- [some-executor](./some-executor/README.md)

## Other sections

Some text
`
    );
  });

  it('should throw when the executor already exists', async () => {
    tree.write(
      'tools/executors/executors.json',
      JSON.stringify({ executors: { [expectedExecutorName]: {} } })
    );

    expect(
      async () => await workspaceExecutorGenerator(tree, { name: executorName })
    ).rejects.toThrow();
  });

  it('should generate files', async () => {
    await workspaceExecutorGenerator(tree, { name: executorName });

    expect(
      tree.exists(`tools/executors/${expectedExecutorName}/impl.ts`)
    ).toBeTruthy();
    expect(
      tree.exists(`tools/executors/${expectedExecutorName}/README.md`)
    ).toBeTruthy();
    expect(
      tree.exists(`tools/executors/${expectedExecutorName}/schema.json`)
    ).toBeTruthy();
  });

  it('should add the executor to the executors.json', async () => {
    await workspaceExecutorGenerator(tree, { name: executorName });

    expect(
      devkit.readJson(tree, `tools/executors/executors.json`)
    ).toMatchSnapshot();
  });

  it('should add the executor to the list of available executors in the README', async () => {
    await workspaceExecutorGenerator(tree, { name: executorName });

    expect(tree.read(`tools/executors/README.md`, 'utf-8')).toMatchSnapshot();
  });

  it('should format files', async () => {
    jest.spyOn(devkit, 'formatFiles');

    await workspaceExecutorGenerator(tree, { name: executorName });

    expect(devkit.formatFiles).toHaveBeenCalled();
  });
});
