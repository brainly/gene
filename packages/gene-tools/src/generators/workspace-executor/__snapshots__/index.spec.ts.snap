// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`workspaceExecutor generator should add the executor to the executors.json 1`] = `
Object {
  "executors": Object {
    "foo-bar": Object {
      "description": "Add a meaningful description for the executor.",
      "implementation": "./foo-bar/impl",
      "schema": "./foo-bar/schema.json",
    },
  },
}
`;

exports[`workspaceExecutor generator should add the executor to the list of available executors in the README 1`] = `
"# Executors

Some description

## Available workspace executors

-   [some-executor](./some-executor/README.md)
-   [foo-bar](foo-bar/README.md): Add a meaningful description for the executor.

## Other sections

Some text
"
`;
